spec/datetimeformat.html

```diff
@@ -1049,23 +1049,23 @@ <h1>PartitionDateTimeRangePattern ( _dateTimeFormat_, _x_, _y_ )</h1>
         The PartitionDateTimeRangePattern abstract operation is called with arguments _dateTimeFormat_ (which must be an object initialized as a DateTimeFormat), _x_ (which must be a Number value) and _y_ (which must be a Number value), interprets _x_ and _y_ as time values as specified in es2023, <emu-xref href="#sec-time-values-and-time-range"></emu-xref>, and creates the corresponding parts according to the effective locale and the formatting options of _dateTimeFormat_. The following steps are taken:
       </p>
       <emu-alg>
-        1. Let _x_ be TimeClip(_x_).
+        1. Set _x_ to TimeClip(_x_).
         1. If _x_ is *NaN*, throw a *RangeError* exception.
-        1. Let _y_ be TimeClip(_y_).
+        1. Set _y_ to TimeClip(_y_).
         1. If _y_ is *NaN*, throw a *RangeError* exception.
         1. Let _tm1_ be ToLocalTime(ℤ(ℝ(_x_) &times; 10<sup>6</sup>), _dateTimeFormat_.[[Calendar]], _dateTimeFormat_.[[TimeZone]]).
         1. Let _tm2_ be ToLocalTime(ℤ(ℝ(_y_) &times; 10<sup>6</sup>), _dateTimeFormat_.[[Calendar]], _dateTimeFormat_.[[TimeZone]]).
         1. Let _rangePatterns_ be _dateTimeFormat_.[[RangePatterns]].
         1. Let _rangePattern_ be *undefined*.
         1. Let _dateFieldsPracticallyEqual_ be *true*.
         1. Let _patternContainsLargerDateField_ be *false*.
-        1. While _dateFieldsPracticallyEqual_ is *true* and _patternContainsLargerDateField_ is *false*, repeat for each row of <emu-xref href="#table-datetimeformat-rangepatternfields"></emu-xref> in order, except the header row:
+        1. For each row of <emu-xref href="#table-datetimeformat-rangepatternfields"></emu-xref>, except the header row, in table order, do
           1. Let _fieldName_ be the name given in the Range Pattern Field column of the row.
           1. If _rangePatterns_ has a field [[&lt;_fieldName_&gt;]], let _rp_ be _rangePatterns_.[[&lt;_fieldName_&gt;]]; else let _rp_ be *undefined*.
           1. If _rangePattern_ is not *undefined* and _rp_ is *undefined*, then
             1. Set _patternContainsLargerDateField_ to *true*.
-          1. Else,
-            1. Let _rangePattern_ be _rp_.
+          1. If _dateFieldsPracticallyEqual_ is *true* and _patternContainsLargerDateField_ is *false*, then
```

Comment:
It was a bit difficult to follow this change but now I'm confident the final result makes the flow simpler, thanks!